sql/atxs/atxs_test.go

```diff
@@ -597,10 +597,7 @@ func TestLoadBlob(t *testing.T) {
 	require.Equal(t, []int{len(blob1.Bytes)}, blobSizes)
 
 	var blob2 sql.Blob
-	atx2, err := newAtx(sig, func(atx *types.ActivationTx) {
-		nodeID := types.RandomNodeID()
-		atx.NodeID = &nodeID // ensure ATXs differ in size
-	})
+	atx2, err := newAtx(sig)
```

Comment:
Is there a way to have different sizes in these ATXs or is the new wire type of constant size?
The test verifies that there's no mistake in the query that could be causing it to e.g. return the same size for different blobs

Reply:
I don't understand, the test passes without this - is it missing some assertions? Perhaps you could clarify what's the purpose of this test, wdym by "return the same size for different blobs"?

Reply:
@ivan4th an ATX can vary in size (i.e. have different number of bytes in it's blob). The reason for this is that it

a) initial ATXs contain 2 posts instead of one
b) initial ATXs (and occasionally non-initial ones) contain a post VRF
c) a PoST has a varying size, i.e. a post for 256 GiB is smaller than a PoST for 4 TiB

Reply:
There's `atxs.GetBlobSizes` func that returns the sizes of multiple blobs: https://github.com/spacemeshos/go-spacemesh/blob/213a58503544b24684acdd29fbffd6376876b996/sql/atxs/atxs.go#L334
which is based on `sql.GetBlobSizes`: https://github.com/spacemeshos/go-spacemesh/blob/213a58503544b24684acdd29fbffd6376876b996/sql/database.go#L527
It seems to me that it's relatively easy to imagine a potential bug in this func (which could be introduced by a future refactor) making it return wrong sizes for the blobs, e.g. same size for each one. Thus, I intentionally used ATXs with different sizes of the serialized representation to check it, as was noted in the comment in the replaced part of code

Reply:
OK; I misunderstood your earlier comment. Yes the test should add ATXs of different sizes and now that I looked at the test closer I noticed a different problem @poszu 

This tests generates a `types.VerifiedActivationTx` that is then signed using `activation.SignAndFinalizeAtx` which also converts the domain type to a wire type, which isn't possible any more - the domain type doesn't contain a post any more.

This results in the blob being incomplete.

In my opinion `sql.Add()` should take 2 arguments - the domain object and a `[]byte` that will be stored as the blob for the ATX with the ID in the domain type. This will probably require quite a few other tests to be updated (but only the `activation.Handler` in non-test code).

Reply:
@ivan4th Thanks for the explanation. I will fix the test and add the missing assertion verifying that the blobs in fact have different size.

> In my opinion atxs.Add() should take 2 arguments - the domain object and a []byte that will be stored as the blob for the ATX with the ID in the domain type. 

It already "takes" the blob which is part of the `types.VerifiedActivationTx`.

Reply:
> It already "takes" the blob which is part of the `types.VerifiedActivationTx.`

Yes, I remember we already had a similar discussion on a previous PR 😅